functions-v2/src/utils.ts

```diff
@@ -0,0 +1,3 @@
+export function isArrayEqual(array1: string[] | undefined, array2: string[]) {
+  return array1?.length === array2.length && array1.every((value, index) => value === array2[index]);
+}
```

Comment:
I'm wondering if it would be simpler/better to use a standard function, like `_.isEqual`.

Reply:
My reluctance to do that is to keep the dependencies used by the functions minimal. 

The effect of dependencies on function start up time is not well documented. I think functions work like this:
When the function is deployed, a cloud process is run to build a docker image for running the function. This cloud process runs `npm install` to get all of the dependencies into the node_modules folder within the docker image. When the function is run the docker image is downloaded onto a virtual server and started up. If the function has been run recently then this docker image will already be present on the server so the startup is faster.

If I'm right about that, the size of the docker image will impact the startup time. Also I think if node has to import a lot of modules before running the function that will be slower.

So the upshot of all that is that unless we use a packaging system that builds a minimal function, I'm going to be reluctant to add dependencies.

Reply:
That makes sense!   It's weird some of the things JS does not have without adding dependencies.